services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchesSpec.groovy

```diff
@@ -195,4 +197,94 @@ class SwitchesSpec extends HealthCheckSpecification {
+        northbound.installSwitchRules(NON_EXISTENT_SWITCH_ID, InstallRulesAction.INSTALL_DEFAULTS)
+
+        then: "Not Found error is returned"
+        def e = thrown(HttpClientErrorException)
+        e.statusCode == HttpStatus.NOT_FOUND
+        e.responseBodyAsString.to(MessageError).errorMessage == "Switch $NON_EXISTENT_SWITCH_ID was not found"
+    }
+
+    @Ignore("is not working")
+    @Tags(LOW_PRIORITY)
+    def "System returns human readable error when deleting switch rules on non-existing switch"() {
+        when: "Delete switch rules on non-existing switch"
+        northbound.deleteSwitchRules(NON_EXISTENT_SWITCH_ID, DeleteRulesAction.DROP_ALL_ADD_DEFAULTS)
+
+        then: "Not Found error is returned"
+        def e = thrown(HttpClientErrorException)
+        e.statusCode == HttpStatus.NOT_FOUND
+        e.responseBodyAsString.to(MessageError).errorMessage == "Switch $NON_EXISTENT_SWITCH_ID was not found."
+    }
+
+    @Tags(LOW_PRIORITY)
+    def "System returns human readable error when setting under maintenance non-existing switch"() {
+        when: "set under maintenance non-existing switch"
+        northbound.setSwitchMaintenance(NON_EXISTENT_SWITCH_ID, true, true)
+
+        then: "Not Found error is returned"
+        def e = thrown(HttpClientErrorException)
+        e.statusCode == HttpStatus.NOT_FOUND
+        e.responseBodyAsString.to(MessageError).errorMessage == "Switch $NON_EXISTENT_SWITCH_ID not found."
+    }
+
+    @Tags(LOW_PRIORITY)
+    def "System returns human readable error when requesting all meters from non-existing switch"() {
+        when: "Request all meters from non-existing switch"
+        northbound.getAllMeters(NON_EXISTENT_SWITCH_ID)
+
+        then: "Not Found error is returned"
+        def e = thrown(HttpClientErrorException)
+        e.statusCode == HttpStatus.NOT_FOUND
+        e.responseBodyAsString.to(MessageError).errorMessage == "Switch $NON_EXISTENT_SWITCH_ID was not found"
+    }
+
+    @Tags(LOW_PRIORITY)
+    def "System returns human readable error when deleting meter on non-existing switch"() {
+        when: "Delete meter on non-existing switch"
+        northbound.deleteMeter(NON_EXISTENT_SWITCH_ID, 33)
+
+        then: "Not Found error is returned"
+        def e = thrown(HttpClientErrorException)
+        e.statusCode == HttpStatus.NOT_FOUND
+        e.responseBodyAsString.to(MessageError).errorMessage == "Switch $NON_EXISTENT_SWITCH_ID was not found"
+    }
+
+    @Tags(LOW_PRIORITY)
+    def "System returns human readable error when configuring port on non-existing switch"() {
+        when: "Configure port on non-existing switch"
+        northbound.portUp(NON_EXISTENT_SWITCH_ID, 33)
+
+        then: "Not Found error is returned"
+        def e = thrown(HttpClientErrorException)
+        e.statusCode == HttpStatus.NOT_FOUND
+        e.responseBodyAsString.to(MessageError).errorMessage == "Switch $NON_EXISTENT_SWITCH_ID was not found"
```

Comment:
I'm fine with how it is now, but could have been done as parameterized test
```
where:
data << [
  [descr: "delete", operation: { northbound.deleteSwitch(...) }],
  [descr: "port up", operation: { northbound.portUp(...) }]
]
```

Reply:
Only @Shared and static fields may be accessed from here.
I will leave it as is

Reply:
just do `getNorthbound()`